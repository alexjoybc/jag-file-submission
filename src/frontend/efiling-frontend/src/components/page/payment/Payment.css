.half-width {
  width: 50%;
}

.court-half-width {
  width: 50%;
}

@media (max-width: 1200px) {
  .half-width {
    width: 80%;
  }
}
