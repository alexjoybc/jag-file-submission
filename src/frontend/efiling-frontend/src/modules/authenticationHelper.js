const jwt = require("jsonwebtoken");

export function getJWTData() {
  const token = localStorage.getItem("jwt");
  if (!token) return false;

  // get the decoded payload and header
  const decoded = jwt.decode(token, { complete: true });
  return decoded.payload;
}

export function generateJWTToken(payload) {
  return jwt.sign(payload, "something");
}
