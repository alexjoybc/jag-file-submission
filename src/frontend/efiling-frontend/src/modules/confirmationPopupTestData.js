/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const mainButton = {
  label: "Cancel",
  styling: "normal-white btn"
};

const confirmButton = {
  label: "Yes, cancel",
  styling: "normal-blue btn consistent-width"
};

const cancelButton = {
  label: "No, dont cancel",
  styling: "normal-white btn consistent-width"
};

const handleClose = () => {};
const handleShow = () => {};

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getTestData() {
  const modal = {
    show: true,
    handleShow,
    handleClose,
    handleConfirm: () => {},
    title: "title",
    body
  };

  return { modal, mainButton, confirmButton, cancelButton };
}
