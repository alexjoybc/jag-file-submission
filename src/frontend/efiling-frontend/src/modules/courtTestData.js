const court = {
  locationDescription: "Court location",
  fileNumber: "Court file number",
  levelDescription: "Level",
  classDescription: "Class",
  level: "P",
  courtClass: "F",
};

export function getCourtData() {
  return court;
}
