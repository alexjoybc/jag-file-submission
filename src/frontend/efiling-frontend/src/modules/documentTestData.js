const documents = [
  {
    name: "file name 1",
    description: "file description 1",
    type: "file type",
    statutoryFeeAmount: 40
  },
  {
    name: "file name 2",
    description: "file description 2",
    type: "file type",
    statutoryFeeAmount: 0
  }
];

export function getDocumentsData() {
  return documents;
}
