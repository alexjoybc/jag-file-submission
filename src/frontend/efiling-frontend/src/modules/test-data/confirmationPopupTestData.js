/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const onButtonClick = () => {};

const mainButton = {
  label: "Cancel",
  styling: "normal-white btn",
  onClick: onButtonClick,
};

const confirmButton = {
  label: "Yes, cancel",
  styling: "normal-blue btn consistent-width",
  onClick: onButtonClick,
};

const cancelButton = {
  label: "No, dont cancel",
  styling: "normal-white btn consistent-width",
  onClick: onButtonClick,
};

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getTestData() {
  const modal = {
    show: false,
    title: "title",
    body,
  };

  return { modal, mainButton, confirmButton, cancelButton };
}
