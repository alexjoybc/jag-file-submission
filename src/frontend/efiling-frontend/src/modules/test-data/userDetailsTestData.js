const userDetails = {
  internalClientNumber: "ABC123",
  accounts: [
    {
      type: "CSO",
      identifier: "123",
    },
  ],
};

export function getUserDetails() {
  return userDetails;
}
