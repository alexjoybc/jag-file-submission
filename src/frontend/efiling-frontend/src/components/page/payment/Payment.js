/* eslint-disable react/jsx-wrap-multilines */
import React, { useState } from "react";
import PropTypes from "prop-types";
import { MdCreditCard } from "react-icons/md";
import ConfirmationPopup, {
  Button,
  Sidecard,
  Alert,
  DisplayBox,
  Table,
  Callout
} from "shared-components";
import { getSidecardData } from "../../../modules/sidecardData";
import { propTypes } from "../../../types/propTypes";
import PackageConfirmation from "../package-confirmation/PackageConfirmation";
import { generateFileSummaryData } from "../../../modules/generateFileSummaryData";

const generateCourtDataTable = ({
  fileNumber,
  locationDescription,
  levelDescription,
  classDescription
}) => {
  const courtElements = [
    {
      name: "Court File Number:",
      value: fileNumber,
      isValueBold: true
    },
    {
      name: "Location:",
      value: locationDescription,
      isValueBold: true
    },
    {
      name: "Level and Class:",
      value: `${levelDescription} ${classDescription}`,
      isValueBold: true
    }
  ];

  return [
    {
      name: (
        <div className="fit-table">
          <Table elements={courtElements} />
        </div>
      ),
      value: "",
      isSideBySide: true
    }
  ];
};

const getTableElements = (files, submissionFee) => {
  const fileSummary = generateFileSummaryData(files, submissionFee, true);

  const elements = [
    {
      name: (
        <div style={{ width: "60%", minWidth: "fit-content" }}>
          <Table isFeesData elements={fileSummary} />
        </div>
      ),
      value: (
        <p>
          The registry will process statutory fees when your documents are
          filed.
        </p>
      ),
      isSideBySide: true
    }
  ];

  return elements;
};

const calloutText = `I have reviewed the information and the documents in this filing
package and am prepared to submit them for filing. I agree that all
fees for this filing package may be charged to the credit card
registered to my account.`;

const submitButton = {
  label: "Submit",
  onClick: () => console.log("submit click"),
  styling: "normal-blue btn"
};

export default function Payment({
  payment: { confirmationPopup, submissionId, courtData, files, submissionFee }
}) {
  const aboutCsoSidecard = getSidecardData().aboutCso;
  const csoAccountDetailsSidecard = getSidecardData().csoAccountDetails;
  const rushSubmissionSidecard = getSidecardData().rushSubmission;
  const [paymentAgreed, setPaymentAgreed] = useState(false);
  const [showPackageConfirmation, setShowPackageConfirmation] = useState(false);

  if (showPackageConfirmation) {
    return (
      <PackageConfirmation
        packageConfirmation={{ confirmationPopup, submissionId }}
        csoAccountStatus={{ isNew: false }}
      />
    );
  }

  return (
    <div className="page">
      <div className="content col-md-8">
        <h1>Payment</h1>
        {/* TODO: Fix credit card info and link to register card */}
        <Alert
          icon={<MdCreditCard size={32} />}
          type="warning"
          styling="warning-background no-padding-bottom"
          element={
            <p>
              Your credit card, BLAH, will be charged.&nbsp;
              <a
                href="https://justice.gov.bc.ca/cso/about/index.do"
                target="_blank"
                rel="noopener noreferrer"
              >
                Register a new Credit Card
              </a>
              .
            </p>
          }
        />
        <br />
        <DisplayBox
          styling="display-left-element"
          element={<Table elements={getTableElements(files, submissionFee)} />}
        />
        <br />
        <h1>Package Submission Details</h1>
        <p>Your package will be filed to:</p>
        <Table elements={generateCourtDataTable(courtData)} />
        <Callout
          text={calloutText}
          checkboxLabel="I agree"
          agreeCallout={() => setPaymentAgreed(!paymentAgreed)}
        />
        <br />
        <section className="inline-block pt-2">
          <Button
            label="< Back"
            onClick={() => setShowPackageConfirmation(true)}
            styling="normal-white btn"
          />
        </section>
        <section className="buttons pt-2">
          <ConfirmationPopup
            modal={confirmationPopup.modal}
            mainButton={confirmationPopup.mainButton}
            confirmButton={confirmationPopup.confirmButton}
            cancelButton={confirmationPopup.cancelButton}
          />
          <Button
            label={submitButton.label}
            onClick={submitButton.onClick}
            styling={submitButton.styling}
            disabled={!paymentAgreed}
          />
        </section>
      </div>
      <div className="sidecard">
        <Sidecard sideCard={rushSubmissionSidecard} />
        <Sidecard sideCard={csoAccountDetailsSidecard} />
        <Sidecard sideCard={aboutCsoSidecard} />
      </div>
    </div>
  );
}

Payment.propTypes = {
  payment: PropTypes.shape({
    confirmationPopup: propTypes.confirmationPopup,
    submissionId: PropTypes.string.isRequired,
    courtData: PropTypes.object.isRequired,
    files: PropTypes.array.isRequired,
    submissionFee: PropTypes.number.isRequired
  }).isRequired
};
