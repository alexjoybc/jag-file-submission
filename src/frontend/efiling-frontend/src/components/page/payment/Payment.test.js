import React from "react";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import {
  render,
  fireEvent,
  getByText,
  getByRole,
  waitFor,
} from "@testing-library/react";
import { getTestData } from "../../../modules/test-data/confirmationPopupTestData";
import { getDocumentsData } from "../../../modules/test-data/documentTestData";
import { getCourtData } from "../../../modules/test-data/courtTestData";
import { generateJWTToken } from "../../../modules/helpers/authentication-helper/authenticationHelper";

import Payment from "./Payment";

describe("Payment Component", () => {
  const confirmationPopup = getTestData();
  const submissionId = "abc123";
  const apiRequest = `/submission/${submissionId}/filing-package`;
  const files = getDocumentsData();
  const submissionFee = 25.5;
  const courtData = getCourtData();
  const payment = {
    confirmationPopup,
    submissionId,
    courtData,
    files,
    submissionFee,
  };

  sessionStorage.setItem("cardRegistered", true);
  const token = generateJWTToken({
    preferred_username: "username@bceid",
    realm_access: {
      roles: ["rush_flag"],
    },
  });
  localStorage.setItem("jwt", token);

  let mock;
  beforeEach(() => {
    mock = new MockAdapter(axios);
    window.open = jest.fn();
  });

  test("Matches the snapshot", () => {
    const { asFragment } = render(<Payment payment={payment} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("Agreeing to conditions toggles button disabled state", () => {
    const { container } = render(<Payment payment={payment} />);

    expect(getByText(container, "Submit").disabled).toBeTruthy();

    fireEvent.click(getByRole(container, "checkbox"));

    expect(getByText(container, "Submit").disabled).toBeFalsy();

    fireEvent.click(getByRole(container, "checkbox"));

    expect(getByText(container, "Submit").disabled).toBeTruthy();
  });

  test("On back click, it redirects back to the package confirmation page", async () => {
    mock.onGet(apiRequest).reply(200, {
      documents: files,
      court: courtData,
      submissionFeeAmount: submissionFee,
    });

    const { container, asFragment } = render(<Payment payment={payment} />);

    fireEvent.click(getByText(container, "< Back"));

    await waitFor(() => {});

    expect(asFragment()).toMatchSnapshot();
  });

  test("Submit on success redirects to success page", async () => {
    sessionStorage.setItem("successUrl", "success.com");

    mock
      .onPost(`/submission/${submissionId}/submit`)
      .reply(200, { transactionId: 1 });

    const { container } = render(<Payment payment={payment} />);

    fireEvent.click(getByRole(container, "checkbox"));
    fireEvent.click(getByText(container, "Submit"));

    await waitFor(() => {});

    expect(window.open).toHaveBeenCalledWith("success.com", "_self");
  });

  test("Submit on error redirects to error page", async () => {
    sessionStorage.setItem("errorUrl", "error.com");

    mock
      .onPost(`/submission/${submissionId}/submit`)
      .reply(400, { message: "There was an error." });

    const { container } = render(<Payment payment={payment} />);

    fireEvent.click(getByRole(container, "checkbox"));
    fireEvent.click(getByText(container, "Submit"));

    await waitFor(() => {});

    expect(window.open).toHaveBeenCalledWith(
      "error.com?status=400&message=There was an error.",
      "_self"
    );
  });

  test("Click on request rush submission opens rush submission page", async () => {
    mock.onGet(apiRequest).reply(200, {
      documents: files,
      court: courtData,
      submissionFeeAmount: submissionFee,
    });

    const { container, asFragment } = render(<Payment payment={payment} />);

    fireEvent.click(getByText(container, "Request rush submission"));

    await waitFor(() => {});

    expect(asFragment()).toMatchSnapshot();
  });
});
