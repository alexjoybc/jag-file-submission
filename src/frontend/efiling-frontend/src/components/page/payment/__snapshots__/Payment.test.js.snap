// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Component Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <div
      class="content col-md-8"
    >
      <h1>
        Payment
      </h1>
      <div
        class="display-box warning-background no-padding-bottom"
      >
        <div
          class="display-icon"
        >
          <div
            style="color: rgb(252, 186, 25);"
          >
            <svg
              fill="currentColor"
              height="32"
              size="32"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4v-6h16v6zm0-10H4V6h16v2z"
              />
            </svg>
          </div>
        </div>
        <div
          class="display-right-element"
        >
          <p>
            Your credit card, BLAH, will be charged. 
            <a
              href="https://justice.gov.bc.ca/cso/about/index.do"
              rel="noopener noreferrer"
              target="_blank"
            >
              Register a new Credit Card
            </a>
            .
          </p>
        </div>
      </div>
      <br />
      <div
        class="display-box display-left-element"
      >
        <div
          class="display-right-element"
        >
          <table
            class=""
          >
            <thead>
              <tr>
                <th
                  colspan="2"
                />
              </tr>
            </thead>
            <tbody>
              <tr
                class=""
                colspan="2"
              >
                <td
                  class="side-by-side"
                >
                  <div
                    style="width: 60%; min-width: fit-content;"
                  >
                    <table
                      class=""
                    >
                      <thead>
                        <tr>
                          <th
                            colspan="2"
                          />
                        </tr>
                      </thead>
                      <tbody>
                        <tr
                          class=""
                          colspan="2"
                        >
                          <td
                            class=""
                          >
                            Number of Documents in Package:
                          </td>
                          <td
                            class="right-align"
                          >
                            <b>
                              2
                            </b>
                          </td>
                        </tr>
                        <tr
                          class=""
                          colspan="2"
                        >
                          <td
                            class=""
                          >
                            Statutory Fees:
                          </td>
                          <td
                            class="right-align"
                          >
                            <b>
                              $40.00
                            </b>
                          </td>
                        </tr>
                        <tr
                          class=""
                          colspan="2"
                        >
                          <td
                            class=""
                          >
                            Submission Fee:
                          </td>
                          <td
                            class="right-align"
                          >
                            <b>
                              $25.50
                            </b>
                          </td>
                        </tr>
                        <tr
                          class="empty-row"
                          colspan="2"
                        >
                          <td
                            class=""
                          />
                          <td />
                        </tr>
                        <tr
                          class=""
                          colspan="2"
                        >
                          <td
                            class=""
                          >
                            Total Fee:
                          </td>
                          <td
                            class="right-align"
                          >
                            <b>
                              $65.50
                            </b>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </td>
                <td>
                  <p>
                    The registry will process statutory fees when your documents are filed.
                  </p>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <br />
      <h1>
        Package Submission Details
      </h1>
      <p>
        Your package will be filed to:
      </p>
      <table
        class=""
      >
        <thead>
          <tr>
            <th
              colspan="2"
            />
          </tr>
        </thead>
        <tbody>
          <tr
            class=""
            colspan="2"
          >
            <td
              class="side-by-side"
            >
              <div
                class="fit-table"
              >
                <table
                  class=""
                >
                  <thead>
                    <tr>
                      <th
                        colspan="2"
                      />
                    </tr>
                  </thead>
                  <tbody>
                    <tr
                      class=""
                      colspan="2"
                    >
                      <td
                        class=""
                      >
                        Court File Number:
                      </td>
                      <td
                        class=""
                      >
                        <b>
                          Court file number
                        </b>
                      </td>
                    </tr>
                    <tr
                      class=""
                      colspan="2"
                    >
                      <td
                        class=""
                      >
                        Location:
                      </td>
                      <td
                        class=""
                      >
                        <b>
                          Court location
                        </b>
                      </td>
                    </tr>
                    <tr
                      class=""
                      colspan="2"
                    >
                      <td
                        class=""
                      >
                        Level and Class:
                      </td>
                      <td
                        class=""
                      >
                        <b>
                          Level Class
                        </b>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </td>
            <td />
          </tr>
        </tbody>
      </table>
      <div
        class="bcgov-callout-checkbox"
      >
        <p>
          I have reviewed the information and the documents in this filing
package and am prepared to submit them for filing. I agree that all
fees for this filing package may be charged to the credit card
registered to my account.
        </p>
        <label
          for="agreeCallout"
          style="float: right;"
        >
          <input
            id="agreeCallout"
            type="checkbox"
          />
           
          <b>
            I agree
          </b>
          <span
            class="mandatory"
            id="asterisk"
          >
            *
          </span>
        </label>
      </div>
      <br />
      <section
        class="inline-block pt-2"
      >
        <div>
          <button
            class="normal-white btn"
            data-test-id=""
            type="button"
          >
            &lt; Back
          </button>
        </div>
      </section>
      <section
        class="buttons pt-2"
      >
        <div>
          <button
            class="normal-white btn"
            data-test-id="main-cancel-btn"
            type="button"
          >
            Cancel
          </button>
        </div>
        <div>
          <button
            class="normal-blue btn"
            data-test-id=""
            disabled=""
            type="button"
          >
            Submit
          </button>
        </div>
      </section>
    </div>
    <div
      class="sidecard"
    >
      <div
        class="wide-dashboard-spacing"
        style="position: relative;"
      >
        <div
          class="row"
        >
          <div
            class="col-12 pt-2"
            style="position: relative;"
          >
            <p />
            <section
              class="bluegrey-container"
              id="bluegrey-section"
            >
              <div
                class="container-background bluegrey-heading"
              >
                <h2
                  class="heading-style"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="side-card-row"
                    >
                      <div
                        class="round-icon-wrapper"
                      >
                        <svg
                          class="side-card-icon"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M15 1H9v2h6V1zm-4 13h2V8h-2v6zm8.03-6.61l1.42-1.42c-.43-.51-.9-.99-1.41-1.41l-1.42 1.42C16.07 4.74 14.12 4 12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9 9-4.03 9-9c0-2.12-.74-4.07-1.97-5.61zM12 20c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
                          />
                        </svg>
                      </div>
                      <div
                        class="side-card-title"
                      >
                        Rush/Urgent Submission
                      </div>
                    </div>
                  </div>
                </h2>
              </div>
              <div
                class="bluegrey-content"
              >
                <span
                  class="content-style"
                >
                  <p>
                    If you wish to request that this package be submitted on an urgent (rush) basis, you must provide a reason for your request to be considered. 
                    <a
                      href="https://justice.gov.bc.ca/cso/about/index.do"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Request rush submission
                    </a>
                    .
                  </p>
                </span>
              </div>
            </section>
          </div>
        </div>
      </div>
      <div
        class="wide-dashboard-spacing"
        style="position: relative;"
      >
        <div
          class="row"
        >
          <div
            class="col-12 pt-2"
            style="position: relative;"
          >
            <p />
            <section
              class="bluegrey-container"
              id="bluegrey-section"
            >
              <div
                class="container-background bluegrey-heading"
              >
                <h2
                  class="heading-style"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="side-card-row"
                    >
                      <div
                        class="round-icon-wrapper"
                      >
                        <svg
                          class="side-card-icon"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                          />
                        </svg>
                      </div>
                      <div
                        class="side-card-title"
                      >
                        Your CSO Account
                      </div>
                    </div>
                  </div>
                </h2>
              </div>
              <div
                class="bluegrey-content"
              >
                <span
                  class="content-style"
                >
                  <p>
                    CSO account 
                    <b />
                     is linked to your Basic BCeID account 
                    <b>
                      blah2
                    </b>
                     and will be used to file documents. 
                    <a
                      href="https://justice.gov.bc.ca/cso/about/index.do"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      View your CSO account details
                    </a>
                    .
                  </p>
                </span>
              </div>
            </section>
          </div>
        </div>
      </div>
      <div
        class="wide-dashboard-spacing"
        style="position: relative;"
      >
        <div
          class="row"
        >
          <div
            class="col-12 pt-2"
            style="position: relative;"
          >
            <p />
            <section
              class="bluegrey-container"
              id="bluegrey-section"
            >
              <div
                class="container-background bluegrey-heading"
              >
                <h2
                  class="heading-style"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="side-card-row"
                    >
                      <div
                        class="round-icon-wrapper"
                      >
                        <svg
                          class="side-card-icon"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
                          />
                        </svg>
                      </div>
                      <div
                        class="side-card-title"
                      >
                        About E-File Submission
                      </div>
                    </div>
                  </div>
                </h2>
              </div>
              <div
                class="bluegrey-content"
              >
                <span
                  class="content-style"
                >
                  <p>
                    E-File submission is a service to help you securely and electronically file documents with the Government of British Columbia Court Services Online (CSO). 
                    <a
                      href="https://justice.gov.bc.ca/cso/about/index.do"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Learn more about CSO
                    </a>
                    .
                  </p>
                </span>
              </div>
            </section>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Payment Component On back click, it redirects back to the package confirmation page 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <div
      class="content col-md-8"
    >
      <h1>
        Package Confirmation
      </h1>
      <span>
        Review your package for accuracy and upload any additional or supporting documents.
      </span>
      <br />
      <span>
        If there are any errors in these documents, please Cancel this process and re-submit.
      </span>
      <br />
      <br />
      <div>
        <div
          class="display-box border-background display-file"
        >
          <div
            class="display-icon"
          >
            <div
              style="color: rgb(252, 186, 25);"
            >
              <svg
                fill="currentColor"
                height="32"
                size="32"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 16H8v-2h8v2zm0-4H8v-2h8v2zm-3-5V3.5L18.5 9H13z"
                />
              </svg>
            </div>
          </div>
          <div
            class="display-right-element"
          >
            <table
              class=""
            >
              <thead>
                <tr>
                  <th
                    colspan="2"
                  />
                </tr>
              </thead>
              <tbody>
                <tr
                  class=""
                  colspan="2"
                >
                  <td
                    class="vertical-middle"
                  >
                    <div
                      style="width: 80%;"
                    >
                      <span
                        class="file-href"
                        role="button"
                        tabindex="0"
                      >
                        file name 1
                      </span>
                    </div>
                  </td>
                  <td>
                    <table
                      class=""
                    >
                      <thead>
                        <tr>
                          <th
                            colspan="2"
                          />
                        </tr>
                      </thead>
                      <tbody>
                        <tr
                          class=""
                          colspan="2"
                        >
                          <td
                            class="reduce-width"
                          >
                            Description:
                          </td>
                          <td
                            class=""
                          >
                            <b>
                              file description 1
                            </b>
                          </td>
                        </tr>
                        <tr
                          class=""
                          colspan="2"
                        >
                          <td
                            class="reduce-width"
                          >
                            Statutory Fee:
                          </td>
                          <td
                            class=""
                          >
                            <b>
                              $40.00
                            </b>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <br />
      </div>
      <div>
        <div
          class="display-box border-background display-file"
        >
          <div
            class="display-icon"
          >
            <div
              style="color: rgb(252, 186, 25);"
            >
              <svg
                fill="currentColor"
                height="32"
                size="32"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 16H8v-2h8v2zm0-4H8v-2h8v2zm-3-5V3.5L18.5 9H13z"
                />
              </svg>
            </div>
          </div>
          <div
            class="display-right-element"
          >
            <table
              class=""
            >
              <thead>
                <tr>
                  <th
                    colspan="2"
                  />
                </tr>
              </thead>
              <tbody>
                <tr
                  class=""
                  colspan="2"
                >
                  <td
                    class="vertical-middle"
                  >
                    <div
                      style="width: 80%;"
                    >
                      <span
                        class="file-href"
                        role="button"
                        tabindex="0"
                      >
                        file name 2
                      </span>
                    </div>
                  </td>
                  <td>
                    <table
                      class=""
                    >
                      <thead>
                        <tr>
                          <th
                            colspan="2"
                          />
                        </tr>
                      </thead>
                      <tbody>
                        <tr
                          class=""
                          colspan="2"
                        >
                          <td
                            class="reduce-width"
                          >
                            Description:
                          </td>
                          <td
                            class=""
                          >
                            <b>
                              file description 2
                            </b>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <br />
      </div>
      <h3>
        Do you have additional documents to upload? 
        <a
          href="#"
        >
          Upload them now.
        </a>
      </h3>
      <br />
      <br />
      <h2>
        Summary
      </h2>
      <br />
      <table
        class=""
      >
        <thead>
          <tr>
            <th
              colspan="2"
            />
          </tr>
        </thead>
        <tbody>
          <tr
            class=""
            colspan="2"
          >
            <td
              class="side-by-side"
            >
              <div
                class="fit-table"
              >
                <table
                  class=""
                >
                  <thead>
                    <tr>
                      <th
                        colspan="2"
                      />
                    </tr>
                  </thead>
                  <tbody>
                    <tr
                      class=""
                      colspan="2"
                    >
                      <td
                        class=""
                      >
                        Number of Documents in Package:
                      </td>
                      <td
                        class=""
                      >
                        <b>
                          2
                        </b>
                      </td>
                    </tr>
                    <tr
                      class=""
                      colspan="2"
                    >
                      <td
                        class=""
                      >
                        Statutory Fees:
                      </td>
                      <td
                        class=""
                      >
                        <b>
                          $40.00
                        </b>
                      </td>
                    </tr>
                    <tr
                      class=""
                      colspan="2"
                    >
                      <td
                        class=""
                      >
                        Submission Fee:
                      </td>
                      <td
                        class=""
                      >
                        <b>
                          $25.50
                        </b>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </td>
            <td />
          </tr>
        </tbody>
      </table>
      <br />
      <section
        class="buttons pt-2"
      >
        <div>
          <button
            class="normal-white btn"
            data-test-id="main-cancel-btn"
            type="button"
          >
            Cancel
          </button>
        </div>
        <div>
          <button
            class="normal-blue btn"
            data-test-id="continue-btn"
            type="button"
          >
            Continue to Payment
          </button>
        </div>
      </section>
    </div>
    <div
      class="sidecard"
    >
      <div
        class="wide-dashboard-spacing"
        style="position: relative;"
      >
        <div
          class="row"
        >
          <div
            class="col-12 pt-2"
            style="position: relative;"
          >
            <p />
            <section
              class="bluegrey-container"
              id="bluegrey-section"
            >
              <div
                class="container-background bluegrey-heading"
              >
                <h2
                  class="heading-style"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="side-card-row"
                    >
                      <div
                        class="round-icon-wrapper"
                      >
                        <svg
                          class="side-card-icon"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                          />
                        </svg>
                      </div>
                      <div
                        class="side-card-title"
                      >
                        Your CSO Account
                      </div>
                    </div>
                  </div>
                </h2>
              </div>
              <div
                class="bluegrey-content"
              >
                <span
                  class="content-style"
                >
                  <p>
                    CSO account 
                    <b />
                     is linked to your Basic BCeID account 
                    <b>
                      blah2
                    </b>
                     and will be used to file documents. 
                    <a
                      href="https://justice.gov.bc.ca/cso/about/index.do"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      View your CSO account details
                    </a>
                    .
                  </p>
                </span>
              </div>
            </section>
          </div>
        </div>
      </div>
      <div
        class="wide-dashboard-spacing"
        style="position: relative;"
      >
        <div
          class="row"
        >
          <div
            class="col-12 pt-2"
            style="position: relative;"
          >
            <p />
            <section
              class="bluegrey-container"
              id="bluegrey-section"
            >
              <div
                class="container-background bluegrey-heading"
              >
                <h2
                  class="heading-style"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="side-card-row"
                    >
                      <div
                        class="round-icon-wrapper"
                      >
                        <svg
                          class="side-card-icon"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
                          />
                        </svg>
                      </div>
                      <div
                        class="side-card-title"
                      >
                        About E-File Submission
                      </div>
                    </div>
                  </div>
                </h2>
              </div>
              <div
                class="bluegrey-content"
              >
                <span
                  class="content-style"
                >
                  <p>
                    E-File submission is a service to help you securely and electronically file documents with the Government of British Columbia Court Services Online (CSO). 
                    <a
                      href="https://justice.gov.bc.ca/cso/about/index.do"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Learn more about CSO
                    </a>
                    .
                  </p>
                </span>
              </div>
            </section>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
